python/lsst/daf/butler/core/storageClass.py

```diff
@@ -160,6 +177,41 @@ def derivedComponents(self) -> Dict[str, StorageClass]:
         """Return derived components associated with `StorageClass`."""
         return self._derivedComponents
 
+    @property
+    def converters(self) -> Dict[str, str]:
+        """Return the type converters supported by this `StorageClass`."""
+        return self._converters
+
+    @property
+    def converters_by_type(self) -> Dict[Type, Type]:
+        """Return the type converters as python types."""
+        if self._converters_by_type is None:
+            self._converters_by_type = {}
+
+            # Loop over list because the dict can be edited in loop.
+            for candidate_type_str, converter_str in list(self.converters.items()):
+                if hasattr(builtins, candidate_type_str):
+                    candidate_type = getattr(builtins, candidate_type_str)
+                else:
+                    try:
+                        candidate_type = doImportType(candidate_type_str)
+                    except ImportError as e:
+                        log.info("Unable to import type %s associated with storage class %s (%s)",
+                                 candidate_type_str, self.name, e)
+                        del self.converters[candidate_type_str]
+                        continue
+
+                try:
+                    converter = doImportType(converter_str)
```

Comment:
I'm not overly enamored by the options here. Checking for `types.FunctionType` seems to be one approach but doesn't cover everything (maybe also check for `types.BuiltInFunctionType`). Seeing if `callable` also works for classes so that's not useful either. You'd find out soon enough when the function falls over later.

Reply:
My worry here was that when exception happens at the call site due to misconfiguration, it's hard to track that exception to a specific `converter_str` in a config. Could be worth to extend diagnostics there with something that says "this was the converter name that you gave us in the config"?

Reply:
Maybe trap for `callable` in the doImportType (to filter out variables) and then also catch the converter call and put in the get_full_type_name in a log message.

Reply:
That should work for most common mistakes. I can imagine more complicated cases, but it's not worth thinking about those at your vacation time before the holidays 🎅

Reply:
Okay. I just pushed a callable check and a `log.error` in the conversion itself to report the converter.